engine/src/main/java/org/destinationsol/game/SolObject.java

```diff
@@ -16,41 +16,187 @@
 
 import java.util.List;
 
+
+/**
+ * Generally speaking, SolObjects are all the game's foreground objects that are in the proximity of player/visible screen.
+ * <p>
+ * {@code SolObject} can be generally crashed into. Background objects aren't usually {@code SolObject}s, but they are
+ * rather just sprites rendered using some background manager. {@code SolObject} that gets too far from player/camera is
+ * usually transformed to {@link FarObject}, which has less runtime executable code and thus is faster and can be
+ * recreated back into {@code SolObject}. Some gameplay-wise unimportant {@code SolObjects}s are also just removed when
+ * they get too far from player/camera, and new ones are generated as needed.
+ */
 public interface SolObject {
-    void update(SolGame game);
 
-    boolean shouldBeRemoved(SolGame game);
+    /**
+     * Called on every game's frame, allows for handling of object-specific runtime needs.
+     * <p>
+     * This method can be used for instance for handling pull of loot towards ships, flickering of lights, aiming
+     * crosses for ships and stuff alike.
+     *
+     * @param game Game this object belongs to.
+     */
+    void update(SolGame game);
 
+    /**
+     * Denotes whether the object should be removed as of the time of calling.
+     * <p>
+     * Object should usually be removed when its health reaches zero, or, for instance, if the object is projectile, it
+     * reaches its target. This method should not handle removal of object for optimization purposes when it gets too
+     * far away, unless the object is specifically designed to exist only in close proximity of player, like some sort
+     * of kinetic defense. Removal for optimization purposes is handled in {@link #toFarObject()}.
+     *
+     * @param game Game this object belongs to.
+     * @return Boolean denoting whether the object should be removed.
+     */
+     boolean shouldBeRemoved(SolGame game);
+
+    /**
+     * Called whenever an object is due to be removed from game.
+     * <p>
+     * This method should handle things like throwing out {@link Loot} when ships are destroyed, or creating smaller
+     * asteroids when a big {@link Asteroid} is destroyed. This method is also responsible for freeing its resources,
+     * when such is applicable. Note that this method is also called when an object is due to be removed for
+     * optimization purposes, and thus you need to explicitly check for {@link #shouldBeRemoved(SolGame)} if you have
+     * some code specific to destruction of object.
+     *
+     * @param game Game this object belongs to.
+     */
     void onRemove(SolGame game);
 
+    /**
+     * Called whenever this object should receive damage.
+     * <p>
+     * Whenever the object is hit by some projectile or other sort of damage dealing thing, this method is called on the
+     * object to allow it to perform its damage handling. This usually means just subtracting the damage from object's
```

Comment:
Please remove the first sentence in this paragraph and note the addition of the missing "the" before "object's":
```
* <p>
* This usually means just subtracting the damage from the object's
```